src/include/duckdb/main/client_context.hpp

```diff
@@ -105,6 +105,9 @@ class ClientContext {
 	void RegisterAppender(Appender *appender);
 	void RemoveAppender(Appender *appender);
 
+	//! Overloading and Instantiating a private function to turn it public without a template type
+	void RunFunctionInTransaction(std::function<void()> &fun);
```

Comment:
Please make this private again, instead, add a `RegisterFunction` to the ClientContext that does what you want it to do.